src/engine/test/source/hlp/hlp_test.cpp

```diff
@@ -120,44 +122,135 @@ TEST(hlpTests, url_parsing)
     ASSERT_EQ("user", result["url.username"]);
 }
 
-TEST(json_test, dummy_test)
+// Test: parsing IP addresses
+TEST(hlpTests_IPaddress, IPaddress_success)
+{
+    const char *logQl = "<source.ip> - <server.ip> -- <source.nat.ip> \"-\" \"-\"";
+    const char *event = "127.0.0.1 - 2001:db8:3333:4444:CCCC:DDDD:EEEE:FFFF -- 0:db8:0:4:CCCC:0:EEEE:FFFF \"-\" \"-\"";
+
+    auto parseOp = getParserOp(logQl);
+    auto result = parseOp(event);
+
+    ASSERT_EQ(true, static_cast<bool>(parseOp));
+    ASSERT_EQ("127.0.0.1", result["source.ip"]);
+    ASSERT_EQ("2001:db8:3333:4444:CCCC:DDDD:EEEE:FFFF", result["server.ip"]);
+    ASSERT_EQ("0:db8:0:4:CCCC:0:EEEE:FFFF", result["source.nat.ip"]);
+}
+
+TEST(hlpTests_IPaddress, IPaddress_failed)
+{
+    const char *logQl = "<server.ip> -";
+    const char *event = ":::DDDD:EEEE: -";
+
+    auto parseOp = getParserOp(logQl);
+    auto result = parseOp(event);
+
+    ASSERT_EQ(true, static_cast<bool>(parseOp));
+    ASSERT_EQ("", result["server.ip"]);
+}
+
+TEST(hlpTests_IPaddress, IPaddress_success_IPV6)
+{
+    const char *logQl = " - <source.nat.ip>";
+    const char *event = " - 2001:db8:3333:AB45:1111:00A:4:1";
+
+    auto parseOp = getParserOp(logQl);
+    auto result = parseOp(event);
+
+    ASSERT_EQ(true, static_cast<bool>(parseOp));
+    ASSERT_EQ("2001:db8:3333:AB45:1111:00A:4:1", result["source.nat.ip"]);
+}
+
+TEST(hlpTests_IPaddress, IPaddress_failed_IPV6)
+{
+    const char *logQl = "<server.ip>";
+    const char *event = "2001:db8:#:$:CCCC:DDDD:EEEE:FFFF";
+
+    auto parseOp = getParserOp(logQl);
+    auto result = parseOp(event);
+
+    ASSERT_EQ(true, static_cast<bool>(parseOp));
+    ASSERT_EQ("", result["server.ip"]);
+}
+
+// Test: parsing JSON objects
+TEST(hlpTests_json, success_1)
 {
     const char *logQl = "<_json1/JSON> - <_json2/JSON>";
     const char *event = "{\"String\":\"This is a string\"} - {\"String\":\"This is another string\"}";
 
     auto parseOp = getParserOp(logQl);
     auto result = parseOp(event);
 
+    ASSERT_EQ(true, static_cast<bool>(parseOp));
     ASSERT_EQ("{\"String\":\"This is a string\"}", result["_json1"]);
     ASSERT_EQ("{\"String\":\"This is another string\"}", result["_json2"]);
 }
 
-TEST(map_test, success_test)
+TEST(hlpTests_json, failed)
```

Comment:
We should rename this test to something like "failed_incomplete_json"

Reply:
done